src/lobby.rs

```diff
@@ -114,7 +136,7 @@ impl Lobby {
         let mut inner = self.inner.lock().await;
         let games = &mut inner.games;
         if let Entry::Occupied(mut entry) = games.entry(game_id) {
-            let participants = entry.get_mut();
+            let participants = &mut entry.get_mut().participants;
```

Comment:
I don't know what `&mut` really does here, I assume gets me a mutable reference to the thing, that should be safe since the inner is locked right? there's no way to get a mutable reference to a whole deep struct?

Reply:
yes, this is correct.

`entry.get_mut()` has type `&mut LobbyGame`
`entry.get_mut().participants` has type `HashSet<Participant>`. You cannot assign it to a variable because that would move the hash set out of the LobbyGame (which would leave the LobbyGame in an invalid state). If `HashSet` was `Copy` you could do this, it would make a copy of the hash set instead of moving it.
`&mut entry.get_mut().participants` has type `&mut HashSet<Participant>`. It lives for as long as `entry.get_mut()` does.